src/CMakeLists.txt

```diff
@@ -123,6 +123,11 @@ if(HAVE_INTEL)
     HAVE_BETTER_YASM_ELF64)
 endif()
 
+if(HAVE_ARM)
+  if(HAVE_ARMV8_SIMD)
```

Comment:
i think we need to update 

https://github.com/ceph/ceph/blob/cb399a9b39f95f352552e155bedfac346ccac0a0/src/CMakeLists.txt#L112-L124

to take `HAVE_ARMV8_SIMD ` into consideration. and then conditionalize on `HAVE_INTEL` and `HAVE_ARMV8_SIMD` in `src/erasure-code/isa/CMakeLists.txt`.

Reply:
it is odd for add HAVE_ARMV8_SIMD under HAVE_INTEL. The src/erasure-code/isa/ is added by src/erasure-code/CMakeLists.txt. I think conditionalize on HAVE_INTEL in src/erasure-code/isa/CMakeLists.txt is inappropriate.

Reply:
> it is odd for add HAVE_ARMV8_SIMD under HAVE_INTEL.

it is. and i don't suggest you to do so.

>The src/erasure-code/isa/ is added by src/erasure-code/CMakeLists.txt. I think conditionalize on HAVE_INTEL in src/erasure-code/isa/CMakeLists.txt is inappropriate.

again, i didn't recommend this.

lemme be more specific. you could change 

https://github.com/ceph/ceph/blob/cb399a9b39f95f352552e155bedfac346ccac0a0/src/CMakeLists.txt#L112-L124

to something like

```cmake
if(HAVE_INTEL OR HAVE_ARMV8_SIMD)
  # ...
```

Reply:
I am sorry ,I am a bit puzzled about this. I think HAVE_BETTER_YASM_ELF64 is for x86 only, it cann't reuse for aarch64. if we consider HAVE_ARMV8_SIMD here we need HAVE_AARCH64_ASM_ELF64  to identify aarch64, It is like below, and then conditionalize on HAVE_BETTER_YASM_ELF64 and HAVE_AARCH64_ASM_ELF64  in src/erasure-code/isa/CMakeLists.txt
```cmake
if(HAVE_INTEL OR HAVE_ARMV8_SIMD)
  if(HAVE_INTEL)
     # ...
     set HAVE_BETTER_YASM_ELF64 1
   elseif(HAVE_ARMV8_SIMD)
     # ...
     set HAVE_AARCH64_ASM_ELF64 1
    endif()
endif()
```
or we just use HAVE_ARMV8_SIMD instead, there is no change in src/CMakelists.txt and 
modify src/erasure-code/CMakeLists.txt like below  and then conditionalize on HAVE_BETTER_YASM_ELF64 and HAVE_ARMV8_SIMD in src/erasure-code/isa/CMakeLists.txt
```cmake
if (HAVE_BETTER_YASM_ELF64 OR HAVE_ARMV8_SIMD )
  add_subdirectory(isa)
  set(EC_ISA_LIB ec_isa)
endif (HAVE_BETTER_YASM_ELF64 OR HAVE_ARMV8_SIMD )